### PR TITLE
Support for @examplesIf roxygen tag

### PR DESCRIPTION
**Pull request by krlmlr**
for #742. Ideally we'd add full support for `@examplesIf`, this gets us halfway. Not sure the final `//s*` is necessary.

Needs tests.
***
Here is how the current PR would change benchmark results when merged into master:

* cache_applying: 0.03 -> 0.03 (-1.3%)
* cache_recording: 1.19 -> 1.19 (-0.4%)
* without_cache: 3.44 -> 3.46 (0.6%)

### CODE REVIEW

**Comment by lorenzwalthert** [comment]
Thanks. We should not match the whole line since you can have `#’ @examples 1+1` and thus is valid (and coveted by unit tests).

File: R/roxygen-examples-find.R [hunk 1]
```diff
@@ -7,7 +7,7 @@
 #' @importFrom rlang seq2
 #' @keywords internal
 identify_start_to_stop_of_roxygen_examples_from_text <- function(text) {
-  starts <- grep("^#'\\s*@examples", text, perl = TRUE)
+  starts <- grep("^#'\\s*@examples\\s*$", text, perl = TRUE)
```
Comment (lorenzwalthert) [inline R/roxygen-examples-find.R:10]:
```suggestion
  starts <- grep("^#'\\s*@examples(\\s*|If\\s|\\t)", text, perl = TRUE)
```

Comment (lorenzwalthert) [inline R/roxygen-examples-find.R:10]:
```suggestion
  starts <- grep("^#'\\s*@examples(If\\s|\\s|\\t)", text, perl = TRUE)
```

File: R/roxygen-examples-find.R [hunk 2]
```diff
@@ -7,7 +7,7 @@
 #' @importFrom rlang seq2
 #' @keywords internal
 identify_start_to_stop_of_roxygen_examples_from_text <- function(text) {
-  starts <- grep("^#'\\s*@examples", text, perl = TRUE)
+  starts <- grep("^#'\\s*@examples(If\\s|\\s|\\t)", text, perl = TRUE)
```
Comment (lorenzwalthert) [inline R/roxygen-examples-find.R:10]:
```suggestion
  starts <- grep("^#'\\s*@examples(If\\s|\\s|\\t|$)", text, perl = TRUE)
```

**Comment by lorenzwalthert** [comment]
@krlmlr you can try this if you want, should work now. Maybe have a look at the test cases too and tell me if we miss anything...

File: R/roxygen-examples-add-remove.R [hunk 3]
```diff
@@ -30,15 +30,29 @@ remove_roxygen_mask <- function(text) {
 #' @keywords internal
 remove_roxygen_header <- function(text) {
   text <- gsub("^\\s*@examples\\s*", "", text, perl = TRUE)
-  starts_with_blank <- text[1] == "\n"
-  c(text[1][!starts_with_blank], text[-1])
+  if (grepl("^If ", text[1])) {
```
Comment (krlmlr) [inline R/roxygen-examples-add-remove.R:33]: This matches `@examples If` . Just sayin...

Can you use the regex from https://github.com/r-lib/styler/pull/743/files#diff-73645d1e1e747d5eeb45cbd47ddb2ef64c658ed5c585763efd004a1b08ba61f9R10, perhaps with rematch2 and a grouped match?

**Review by krlmlr — COMMENTED** [review]
Looks good, thanks for the effort! I didn't review all the examples, will file new issues if anything breaks in practice.

File: R/roxygen-examples-add-remove.R (hunk 3 above)
Comment (lorenzwalthert) [reply R/roxygen-examples-add-remove.R:33]: Good catch. I didn't ever figure out how to use `{rematch2}` (and did not bother spending time), maybe I should... For now, I think using the regex you linked should be sufficient.

**Pull request merged**